sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java

```diff
@@ -563,77 +566,29 @@ private void validateAuthorizationClientProperties(String registrationId,
-                        + "'on_behalf_of'.", registrationId);
-                    break;
-                case WEB_APPLICATION_AND_RESOURCE_SERVER:
-                    throw new IllegalStateException("spring.cloud.azure.active-directory.authorization-clients." + registrationId
-                        + ".authorization-grant-grantType must be configured. ");
-                default:
-                    throw new IllegalStateException("Unsupported authorization grantType " + applicationType.getValue());
-            }
+            setDefaultValueForAuthorizationWhenGrantTypeIsNull(registrationId, properties);
         } else {
             // Validate authorization grant grantType
-            switch (applicationType) {
-                case WEB_APPLICATION:
-                    if (ON_BEHALF_OF.getValue().equals(grantType)) {
-                        throw new IllegalStateException(String.format(UNMATCHING_OAUTH_GRANT_TYPE_FROMAT,
-                            "web_application", registrationId, "on_behalf_of"));
-                    }
-                    break;
-                case RESOURCE_SERVER:
-                    if (AUTHORIZATION_CODE.getValue().equals(grantType)) {
-                        throw new IllegalStateException(String.format(UNMATCHING_OAUTH_GRANT_TYPE_FROMAT,
-                            "resource_server", registrationId, "authorization_code"));
-                    }
-                    if (ON_BEHALF_OF.getValue().equals(grantType)) {
-                        throw new IllegalStateException(String.format(UNMATCHING_OAUTH_GRANT_TYPE_FROMAT,
-                            "resource_server", registrationId, "on_behalf_of"));
-                    }
-                    break;
-                case RESOURCE_SERVER_WITH_OBO:
-                    if (AUTHORIZATION_CODE.getValue().equals(grantType)) {
-                        throw new IllegalStateException(String.format(UNMATCHING_OAUTH_GRANT_TYPE_FROMAT,
-                            "resource_server_with_obo", registrationId, "authorization_code"));
-                    }
-                    break;
-                case WEB_APPLICATION_AND_RESOURCE_SERVER:
-                default:
-                    LOGGER.debug("'spring.cloud.azure.active-directory.authorization-clients." + registrationId
-                        + ".authorization-grant-type' is valid.");
-            }
-
-            if (AZURE_CLIENT_REGISTRATION_ID.equals(registrationId)
-                && AUTHORIZATION_CODE != properties.getAuthorizationGrantType()) {
-                throw new IllegalStateException("spring.cloud.azure.active-directory.authorization-clients."
-                    + AZURE_CLIENT_REGISTRATION_ID
-                    + ".authorization-grant-type must be configured to 'authorization_code'.");
-            }
+            validateAuthorizationGrantType(registrationId, grantType, properties);
```

Comment:
Can we define a matrix to define the unmatching application type and grant types?